### PR TITLE
Using graph api

### PR DESCRIPTION
**Pull request by outscale-toa**

### CODE REVIEW

File: include/packetgraph/vhost.h [hunk 1]
```diff
@@ -52,7 +52,7 @@
 
 #define PG_VHOST_USER_CLIENT		(1ULL << 0)
 #define PG_VHOST_USER_NO_RECONNECT	(1ULL << 1)
-#define PG_VHOST_USER_DEQUEUE_ZERO_COPY	(1ULL << 2)
+#define PG_VHOST_USER_DEQUEUE_ZERO_COPY	 0
```
Comment (outscale-jju) [inline include/packetgraph/vhost.h]: No, you have to change in the test file, not in the API !

File: tests/integration/tests.c [hunk 2]
```diff
@@ -104,7 +104,6 @@ static void pg_gc_destroy(GList *graph)
 	g_list_free(graph);
 }
 
-
```
Comment (outscale-mgo) [inline tests/integration/tests.c]: ?

File: tests/integration/tests.c [hunk 3]
```diff
@@ -113,20 +112,19 @@ static void pg_gc_destroy(GList *graph)
...
-
 #define CHECK_ERROR_ASSERT(error) do {		\
 		if (error) {			\
 			pg_error_print(error);	\
 			ASSERT(0);		\
 		}				\
 	} while (0)
 
+static void check_error(struct pg_error *error)
```
Comment (outscale-mgo) [inline tests/integration/tests.c]: fine, but what's the point ?

File: Makefile.am [hunk 4]
```diff
@@ -68,7 +68,7 @@ pkginclude_HEADERS = \
 libpacketgraph_la_LIBADD = $(RTE_SDK_LIBS) $(GLIB_LIBS)
 # FIXME '^pg_[^_]' does not take all symbols needed (i.e. __pg_error_*)
 libpacketgraph_la_LDFLAGS = -version-info 17:2:0 -export-symbols-regex 'pg_[^_]' -no-undefined
-libpacketgraph_la_CFLAGS = -march=core-avx-i -mtune=core-avx-i -fmessage-length=0 -Werror -Wall -Wextra -Wwrite-strings -Winit-self -Wpointer-arith -Wstrict-aliasing -Wformat=2 -Wmissing-declarations -Wmissing-include-dirs -Wno-unused-parameter -Wuninitialized -Wold-style-definition -Wstrict-prototypes -Wmissing-prototypes -fPIC -std=gnu11 $(GLIB_CFLAGS) $(RTE_SDK_CFLAGS) -I$(srcdir)/include -I$(srcdir)/src -Wimplicit-fallthrough=0 -Wno-unknown-warning-option
+libpacketgraph_la_CFLAGS = -march=core-avx-i -mtune=core-avx-i -fmessage-length=0 -Werror -Wall -Wextra -Wwrite-strings -Winit-self -Wpointer-arith -Wstrict-aliasing -Wformat=2 -Wmissing-declarations -Wmissing-include-dirs -Wno-unused-parameter -Wuninitialized -Wold-style-definition -Wstrict-prototypes -Wmissing-prototypes -fPIC -std=gnu11 $(GLIB_CFLAGS) $(RTE_SDK_CFLAGS) -I$(srcdir)/include -I$(srcdir)/src -Wno-fallthrough=0 -Wno-unknown-warning-option
```
Comment (outscale-mgo) [inline Makefile.am]: yes, -Wno-fallthrough=0 seems wrong, have you try '-Wno-fallthrough' without the = ?
have you try this on clang ?

File: tests/integration/tests.c [hunk 5]
```diff
@@ -331,31 +294,31 @@ static void test_graph_type1(void)
...
-
-	PG_GC_CHAINED_ADD(brick_gc, nic, vtep, print);
-
 	g_assert(add_graph_branch(&branch1, 1, mac1, 0, 0));
 	g_assert(add_graph_branch(&branch2, 2, mac2, 0, 0));
 
 	/*
 	 * Our main graph:
-	 *	                FIREWALL1 -- ANTISPOOF1 -- VHOST1
+	 *	                FIREWALL1 -- VHOST1
```
Comment (outscale-mgo) [inline tests/integration/tests.c]: Can you add a todo please ?
we don't use antispoof yet, but we should use them someday, so please add a todo

File: tests/integration/tests.c [hunk 6]
```diff
@@ -331,31 +294,31 @@ static void test_graph_type1(void)
...
 	g_assert(link_graph_branch(&branch1, vtep));
 	g_assert(link_graph_branch(&branch2, vtep));
 
-	/* Translate MAC to strings */
+	graph = pg_graph_new("graph_1", nic, &error);
+	check_error(error);
+	graph1 = pg_graph_new("graph_2", branch1.vhost_reader, &error);
+	check_error(error);
 
+	/* Translate MAC to strings */
```
Comment (outscale-mgo) [inline tests/integration/tests.c]: no you are not translating mac to string here :)

File: tests/integration/tests.c [hunk 7]
```diff
@@ -367,11 +330,12 @@ static void test_graph_type1(void)
 	g_assert(g_file_test(glob_vm_path, G_FILE_TEST_EXISTS));
 	g_assert(g_file_test(glob_vm_key_path, G_FILE_TEST_EXISTS));
 	g_assert(g_file_test(glob_hugepages_path, G_FILE_TEST_EXISTS));
+
```
Comment (outscale-mgo) [inline tests/integration/tests.c]: space invader here ...

File: tests/integration/tests.c [hunk 8]
```diff
@@ -422,22 +384,23 @@ static void test_graph_type1(void)
...
 	/* check the collect1 */
-	for (int i = 0; i < 10; i++) {
-		usleep(100000);
-		pg_brick_poll(branch1.vhost_reader, &count, &error);
-		CHECK_ERROR_ASSERT(error);
-		if (count)
-			break;
-	}
-
+	g_assert(pg_graph_poll(graph1, &error) == 0);
```
Comment (outscale-mgo) [inline tests/integration/tests.c]: I'm surprise this code work, the usleep was important here

File: tests/integration/tests.c [hunk 9]
```diff
@@ -422,22 +384,23 @@ static void test_graph_type1(void)
...
 					      &error);
 	ASSERT(result_pkts);
+	/* split graph */
+	graph2 = pg_graph_split(graph, "graph-2", "print", "vt",
+						"q_west", "q_east", &error);
+	check_error(error);
+	/* merge graph after splitted */
+	g_assert(pg_graph_merge(graph, graph2, &error) == 0);
+	check_error(error);
+	g_assert(pg_graph_poll(graph, &error) == 0);
```
Comment (outscale-mgo) [inline tests/integration/tests.c]: well that's cool you test this but:
- that's completely unrelated to the pr
- same as before, you remove the usleep...

File: tests/integration/tests.c [hunk 10]
```diff
@@ -517,45 +479,47 @@ static void test_graph_firewall_intense(void)
...
-			     NULL, &error);
-	check_error(error);
-	PG_GC_CHAINED_ADD(brick_gc, nic, vtep, print);
-
-	pg_brick_chained_links(&error, nic, print, vtep);
-	check_error(error);
 
 	for (int i = 0; i < 100; ++i) {
 		for (int j = 0; j < PG_BRANCHES_NB; j++) {
-			g_assert(add_graph_branch(&branches[j], j, mac1, 0, 0));
```
Comment (outscale-mgo) [inline tests/integration/tests.c]: I'm pretty sure you can use add_graph_branch here, it's link_graph_branch that doesn't work with graph.

**Pull request merged**